### PR TITLE
Implement IntoResponse for app Error

### LINKED ISSUE #5: impl axum's IntoResponse for Error
Today the error handling in the controllers is very repetitive. E.g.:
```rust
let list = app_state
        .product_service()
        .list(limit, offset)
        .await
        .map_err(|e| (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()))?;
```
It will be very handy if Error implements IntoResponse. It would become:
```rust
let list = app_state
        .product_service()
        .list(limit, offset)
        .await?;
```

### PR DESCRIPTION
**Pull request by vaniusrb**
Error now implements axum IntoResponse.
Controllers were simplified, returning our app Error.

Fixes #5

### CODE REVIEW

File: src/app/error.rs [hunk 1]
```diff
@@ -7,3 +11,13 @@ pub enum Error {
     #[error("validation error: `{0}`")]
     Validation(String),
 }
+
+impl IntoResponse for Error {
+    fn into_response(self) -> Response {
+        match self {
+            Error::Database(e) => (StatusCode::INTERNAL_SERVER_ERROR, e.to_string()),
+            Error::Validation(e) => (StatusCode::BAD_REQUEST, e),
```
Comment (gustavosimon) [inline src/app/error.rs:19]: Why DatabaseError calls to_string and ValidationError does not?

```rust 
 (StatusCode::INTERNAL_SERVER_ERROR, e.to_string())
```

```rust 
 (StatusCode::BAD_REQUEST, e)
```

Is there an especific reason?

Comment (vaniusrb) [reply src/app/error.rs:19]: The return type of the match should be `(StatusCode, String)`. In that case, the enum's variants are `Error::Database(sqlx::error::Error)` and `Error::Validation(String)`. For the first one, we need to convert `sqlx::error::Error` to `String`. The other already has a `String` value.

Comment (gustavosimon) [reply src/app/error.rs:19]: I got it. That's nice.

**Review by gustavosimon — APPROVED** [review]
LGTM

**Comment by vaniusrb** [comment]
Implemented IntoResponse for app Error

**Pull request merged**